src/actions/contact-request/saveContactRequest.ts

```diff
@@ -1,21 +1,25 @@
 'use server'
-import { createContactRequest } from '@/backend/contact-request/contact-request.service'
+import {
+  createContactRequest,
+  findExistingContactRequest,
+} from '@/backend/contact-request/contact-request.service'
 import { ContactFormRequest } from '@/components/ContactForm/schema'
 
 export const saveContactRequest = async (
   contactFormRequestData: ContactFormRequest,
 ) => {
   try {
-    const createdContactRequest = await createContactRequest(
-      contactFormRequestData,
-    )
-    if (createdContactRequest) {
-      createdContactRequest
-    } else {
+    const existingContactRequest = await findExistingContactRequest({
+      senderEmail: contactFormRequestData.senderEmail,
+      profileId: contactFormRequestData.profileId,
+    })
+    if (existingContactRequest) {
       return false
```

Comment:
**nit:** As I wrote in example - just return `createdContactRequest` when possible. Boolean is not helpful in any way here. Just throw an error when we cannot create new contact request which will be displayed in toast, so:

```ts
if (existingContactRequest) {
  throw Error("You already contacted this dev")
}
```